include/rapidjson/reader.h

```diff
@@ -1483,14 +1484,76 @@ class GenericReader {
         // Parse minus
         bool minus = Consume(s, '-');
 
-        // Parse int: zero / ( digit1-9 *DIGIT )
+        // Parse int: zero / hexadecimal / ( digit1-9 *DIGIT )
         unsigned i = 0;
         uint64_t i64 = 0;
         bool use64bit = false;
+        bool hex = false;
         int significandDigit = 0;
+        // Parse hxadecimal
         if (RAPIDJSON_UNLIKELY(s.Peek() == '0')) {
-            i = 0;
             s.TakePush();
+            if (RAPIDJSON_UNLIKELY(kParseHexadecimalsFlag & parseFlags))  {
+                static int asciiHexToInt[] =
```

Comment:
may use `static const signed char` to reduce memory/space and indication of constness.